clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -2885,7 +2885,8 @@ static void ParseLangArgs(LangOptions &Opts, ArgList &Args, InputKind IK,
   Opts.Coroutines = Opts.CPlusPlus20 || Args.hasArg(OPT_fcoroutines_ts);
 
   Opts.ConvergentFunctions = Opts.OpenCL || (Opts.CUDA && Opts.CUDAIsDevice) ||
-    Args.hasArg(OPT_fconvergent_functions);
+                             (Opts.SYCL && Opts.SYCLIsDevice) ||
```

Comment:
Do we need to check `SYCL` and `SYCLIsDevice` both?

Reply:
Updated check.